### DISCUSSION TITLE
Updating hiem with older transitive dependency on nix

### DISCUSSION THREAD

**Discussion by froismo** [vectordotdev/vector#15616] [Q&A] [answered]
I'm new to rust, so maybe I'm mistaken, and if so, sorry for wasting your time.

Vector depends on [nix version 0.26.1](https://github.com/vectordotdev/vector/blob/master/Cargo.toml#L343), and a Vector-supported version of heim [depends on ^0.23.0](https://github.com/vectordotdev/heim/blob/update-nix/heim-net/Cargo.toml#L22).  When vector is built will the heim module have its own version of nix, or will it use the nix from the enclosing project?

Specifically, I'm hoping to get nix updated to remediate https://github.com/nix-rust/nix/issues/1762 which is in nix versions above 0.25.0.

**Comment by jszwedko** [comment, marked as answer]
Hi @froismo !

(converted the issue you opened to a discussion since I think that fits better)

Vector depends on several versions of nix:

```
  nix@0.23.1
  nix@0.24.2
  nix@0.26.1
```

You can use `cargo tree -i -p nix@some_version` to see the inverted dependency tree. As you note, `heim` depends on 0.23:

```
nix v0.23.1
├── heim-common v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   ├── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   │   └── vector v0.27.0 (/Users/jesse.szwedko/workspace/vector)
│   ├── heim-cpu v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   │   └── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
│   ├── heim-disk v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   │   └── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
│   ├── heim-host v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   │   └── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
│   ├── heim-memory v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│   │   └── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
│   └── heim-net v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c)
│       └── heim v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
└── heim-net v0.1.0-rc.1 (https://github.com/vectordotdev/heim.git?branch=update-nix#76fa765c) (*)
```

Heim will need to be forked and updated to resolve that.

**Reply by jszwedko** [reply]
Well, I see we are already forking it, so you could just open a PR to https://github.com/vectordotdev/heim 😄

**Reply by froismo** [reply]
I'll give it a go, thank you.

**Reply by froismo** [reply]
I had hoped this was a matter of dependabot not running on vector's heim repo, and just needing a little tickle.  It's more than that, and I'm not proficient in rust.  I agree with your [assessment](https://github.com/heim-rs/heim/issues/364#issuecomment-1212543340) heim-rs looks a little dead.

**Comment by wgzuwg** [comment]
这是一封自动回复邮件。已经收到您的来信，我会尽快回复。
